lib/inbox.rb

```diff
@@ -114,11 +114,15 @@ def url_for_path(path)
     end
 
     def url_for_authentication(redirect_uri, login_hint = '', options = {})
-      trialString = 'false'
-      if options[:trial] == true
-        trialString = 'true'
-      end
-      "https://#{@service_domain}/oauth/authorize?client_id=#{@app_id}&trial=#{trialString}&response_type=code&scope=email&login_hint=#{login_hint}&redirect_uri=#{redirect_uri}"
+      arguments = {
+        trial: false,
+        login_hint: login_hint,
+        client_id: @app_id,
+        response_type: 'code',
+        scpe: 'email',
+        redirect_uri: redirect_uri
+      }.merge(options)
+      URI::HTTPS.build(host: "#{@service_domain}/oauth/authorize", query: arguments.to_query).to_s
```

Comment:
I may be wrong, but I think `Hash.to_query` comes from activesupport, not native Ruby.